### PR TITLE
[API CONFORMANCE] Modify API conformance suite for SW plugins

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *Move some properties from mandatory to optional for sw plugins*
 - *...*

### Tickets:
 - *[133459](https://jira.devtools.intel.com/browse/CVS-133459)*

### CODE REVIEW

**Review by sbalandi — APPROVED** [review]

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/ov_plugin/properties.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ INSTANTIATE_TEST_SUITE_P(ov_plugin_mandatory, OVPropertiesIncorrectTests,
 
 const std::vector<ov::AnyMap> default_properties = {
         {},
-        {ov::enable_profiling(true)},
+        {ov::enable_profiling(sw_plugin_in_target_device(ov::test::utils::target_device) ? false : true)},
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/ov_plugin/properties.cpp:27]: I think HW Plugin should have default value false as well.

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 2]
```diff
@@ -210,18 +210,26 @@ TEST_P(OVSetPropComplieModleGetPropTests, SetPropertyAndComplieModelWithPropsWor
...
 
-std::vector<ov::AnyMap> OVPropertiesTestsWithCompileModelProps::getROMandatoryProperties() {
+std::vector<ov::AnyMap> OVPropertiesTestsWithCompileModelProps::getROMandatoryProperties(bool is_sw_device) {
     std::vector<ov::AnyMap> res;
-    res.push_back({{ov::PropertyName(ov::device::full_name.name(), ov::device::full_name.mutability), nullptr}});
-    res.push_back({{ov::PropertyName(ov::device::architecture.name(), ov::device::architecture.mutability), nullptr}});
-    res.push_back({{ov::PropertyName(ov::device::type.name(), ov::device::type.mutability), nullptr}});
-    res.push_back({{ov::PropertyName(ov::execution_devices.name(), ov::execution_devices.mutability), nullptr}});
+    if (!is_sw_device) {
+        res.push_back({{ov::PropertyName(ov::device::full_name.name(), ov::device::full_name.mutability), nullptr}});
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:216]: ov::device::full_name is mandatory for SW plugin.
ov::available_devices and ov::num_streams are not excluded?

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/ov_plugin/properties.cpp (hunk 1 above)
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/ov_plugin/properties.cpp]: Correct. Agreed. Fixed

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp (hunk 2 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp]: Yes, I excluded these properties from mandatory scope for SW plugins

**Review by peterchen-intel — APPROVED** [review]

**Pull request merged**